performance-tracking-core/src/test/java/com/rakuten/tech/mobile/perf/core/MeasurementBufferSpec.java

```diff
@@ -34,38 +25,84 @@
   }
 
   @Test public void shouldProvideNullWhenBufferIsFull() {
-    for (int i = 0; i < buffer.at.length; i++) {
-      buffer.next();
-    }
+    fillBuffer(buffer, 512);
+
     assertThat(buffer.next()).isNull();
   }
 
-  @Test public void shouldNotFailOnNegativeIndex() {
-    buffer.nextTrackingId.set(-5);
-    Measurement next = buffer.next();
-    assertThat(next.trackingId).isEqualTo(-5);
+  @Test public void shouldProvideSequentialIdAfterFullBufferHasCleared() {
+    fillBuffer(buffer, 512);
+
+    buffer.next();
+
+    for (int i = 1; i <= 512; i++) {
+      buffer.getByTrackingId(i).clear();
+    }
+
+    assertThat(buffer.next().trackingId).isEqualTo(513);
+  }
+
+  @Test public void shouldNotUseZeroId() {
+    fillBuffer(buffer, 512);
+
+    assertThat(buffer.getByTrackingId(0)).isNull();
   }
 
-  @Test public void shouldNotFailOnZeroIndex() {
-    buffer.nextTrackingId.set(0);
+  @Test public void shouldUseNegativeIdWhenIntegerOverflowOccurs() {
+    MeasurementBuffer negativeBuffer = new MeasurementBuffer(Integer.MAX_VALUE + 1);
```

Comment:
if we only use this constructor in this test could we just loop `Integer.MAX_VALUE + 1` times doing a `next().clear()` ?

- [ ] Consider removing constructor only used for testing

Reply:
I tried looping to Integer.MAX_VALUE, but it took a very long time to run the test :(

Reply:
yeah that's what i expected.... too bad :(